src/algorithms/toolbox.jl

```diff
@@ -96,18 +96,23 @@ function MPSKit.correlation_length(peps::InfinitePEPS, env::CTMRGEnv; num_vals=2
 end
 
 """
-    product_peps(peps_args...; unitcell=(1, 1), noise_amp=1e-2)
+    product_peps(peps_args...; unitcell=(1, 1), noise_amp=1e-2, basis_fun=ran, element_fun=randn)
```

Comment:
> Do we really want to normalize the peps?

I just meant normalize as dividing by the array norm, not the physical norm, just to keep the numerical values in a reasonable range. I would also be against physically normalizing the PEPS, that would indeed be annoying to handle, so I would just leave as it is right now.

> I like that there is the option to select the basis for the product state, but it might be more physically relevant if we just allow a single-site-state vector, something like `[sin(X), cos(X)]`

That would be nice indeed, but only `[1, 0]` or `[0, 1]` would actually lead to a product state, right? Of course we could just generalize the function to also include more general states, not just strictly product states.

Reply:
Any input vector would lead to a product state though, it's not the amount of nonzero values that determines that, it's the bonddim, which is 1: a vector really is just 2x1x1x1x1.

(You can think about it as there being a local basis change to the [1 0] state, so for example the plus state ++++ is still a product state)

I didn't know that norm(psi) gives the tensor norm, not the state norm. I am not sure I like that ;)

Reply:
Sorry, got a bit confused about the product state thing but of course you're right. Now it should be implemented such that one can generate a single site vector from a kwarg function.

> I didn't know that norm(psi) gives the tensor norm, not the state norm. I am not sure I like that, but I guess it's necessary for the optimization routine?

Well, for the state norm you also need the CTM environment, that's why `norm(psi, env)` gives the state norm. I don't think `norm(psi)` is needed for the optimization (just some `inner` is needed) but `norm(psi)` is rather used to normalize the PEPS tensors during CTMRG so that the tensor elements do not diverge.

Reply:
I am finally starting to look at Pepskit. It’s about time ;-). For generating a random initial PEPS, you might consider an isometric PEPS. Then the state (and tensor) norms are fixed to 1.

Reply:
I guess for now it doesn't really matter too much, as we will "normalize" anyways during the optimization, so it does not make too much sense to invest in that.